core/src/models/Export.ts

```diff
@@ -234,21 +289,29 @@ export class Export extends Model {
         .value
     );
 
-    const _exports = await Export.findAll({
-      where: {
-        mostRecent: false,
-        createdAt: {
-          [Op.lt]: Moment().subtract(days, "days").toDate(),
-        },
-      },
-      order: [["createdAt", "desc"]],
-      limit,
-    });
-
-    for (const i in _exports) {
-      await _exports[i].destroy();
-    }
+    const whereDate = Moment()
+      .subtract(days, "days")
+      .format("YYYY-MM-DD HH:mm:ss");
+
+    const count = await api.sequelize.query(
+      `
+      DELETE FROM exports
+      WHERE id IN (
+        SELECT id FROM exports
+        WHERE "createdAt" < '${whereDate}'
+        AND "createdAt" < (
+          SELECT max("createdAt")
+          FROM exports e2
+          WHERE
+            e2."profileId" = exports."profileId"
+            AND state = 'complete'
+        )
+        LIMIT ${limit}
+      )
+      `,
+      { type: QueryTypes.DELETE }
+    );
```

Comment:
This is interesting - in SQL alone we can delete all old exports, but leave the latest complete one.